File: Problems/Food Delivery System/Java/restaurant/Restaurant.java
package restaurant;

import java.util.List;

public class Restaurant {
    private int id;
    private String name;
    private List<Item> menu;
    private boolean isOpen;

    public Restaurant(int id, String name, List<Item> menu, boolean isOpen) {
        this.id = id;
        this.name = name;
        this.menu = menu;
        this.isOpen = isOpen;
    }

    // Getters and Setters
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public List<Item> getMenu() {
        return menu;
    }

    public void setMenu(List<Item> menu) {
        this.menu = menu;
    }

    public boolean isOpen() {
        return isOpen;
    }

    public void setOpen(boolean open) {
        isOpen = open;
    }

    public void closeItem(int itemId) {
        for (Item item : menu) {
            if (item.getId() == itemId) {
                item.setAvailable(false);
                break;
            }
        }
    }

    public Item getItem(int itemId) {
        for (Item item : menu) {
            if (item.getId() == itemId) {
                return item;
            }
        }
        return null;
    }
}

File: Problems/Food Delivery System/Golang/restaurant/restaurant.go
package restaurant

// Restaurant represents a restaurant with a menu and open/close state.
type Restaurant struct {
	ID     int
	Name   string
	Menu   []Item
	IsOpen bool
}

func NewRestaurant(id int, name string, menu []Item, isOpen bool) *Restaurant {
	return &Restaurant{ID: id, Name: name, Menu: menu, IsOpen: isOpen}
}

func (r *Restaurant) GetItem(itemID int) *Item {
	for i := range r.Menu {
		if r.Menu[i].ID == itemID {
			return &r.Menu[i]
		}
	}
	return nil
}

func (r *Restaurant) CloseItem(itemID int) {
	for i := range r.Menu {
		if r.Menu[i].ID == itemID {
			r.Menu[i].IsAvailable = false
			return
		}
	}
}
